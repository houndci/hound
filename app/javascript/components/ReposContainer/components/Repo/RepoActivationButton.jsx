import React from 'react';

export default class RepoActivationButton extends React.Component {
  get disabledState() {
    const { isProcessingId, repo } = this.props;
    const { id } = repo;

    return (isProcessingId === id) ? "disabled" : null;
  }

  render() {
    const { repo, onRepoClicked } = this.props;
    const { admin, id } = repo;

    if (admin) {
      return (
        <button
          className="repo-toggle"
          onClick={() => onRepoClicked(id)}
          disabled={this.disabledState}
        >
          Activate
        </button>
      );
    } else {
      return (
        <div className="repo-restricted">
          Only repo admins can activate
        </div>
      );
    }
  }
}
