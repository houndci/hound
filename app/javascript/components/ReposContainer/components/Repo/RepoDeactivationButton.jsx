import React from 'react';

export default class RepoDeactivationButton extends React.Component {
  constructor() {
    super();

    this.state = { buttonText: "Active" };
  }

  onMouseOutButton() {
    this.setState({ buttonText: "Active" });
  }

  onMouseOverButton() {
    this.setState({ buttonText: "Deactivate" });
  }

  getDisabledState() {
    const { isProcessingId, repo } = this.props;
    const { id } = repo;

    return (isProcessingId === id) ? "disabled" : null;
  }

  render() {
    const { buttonText } = this.state;
    const { repo, onRepoClicked } = this.props;
    const { admin, id } = repo;

    if (admin) {
      return (
        <button
          className="repo-toggle"
          disabled={this.getDisabledState()}
          onClick={() => onRepoClicked(id)}
          onMouseOut={() => this.onMouseOutButton()}
          onMouseOver={() => this.onMouseOverButton()}
        >
          {buttonText}
        </button>
      );
    } else {
      return (
        <div className="repo-restricted">
          Only repo admins can activate
        </div>
      );
    }
  }
}
