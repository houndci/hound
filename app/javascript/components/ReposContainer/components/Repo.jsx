import React from 'react';
import classNames from 'classnames';

import RepoActivationButton from './Repo/RepoActivationButton';
import RepoDeactivationButton from './Repo/RepoDeactivationButton';

export default class Repo extends React.Component {
  renderButton() {
    const { isProcessingId, repo, onRepoClicked } = this.props;
    const { active } = repo;

    if (active) {
      return (
        <RepoDeactivationButton
          repo={repo}
          onRepoClicked={onRepoClicked}
          isProcessingId={isProcessingId}
        />
      );
    } else {
      return (
        <RepoActivationButton
          repo={repo}
          onRepoClicked={onRepoClicked}
          isProcessingId={isProcessingId}
        />
      );
    }
  }

  render() {
    const { isProcessingId, repo } = this.props;
    const { active, id, name, price_in_cents } = repo;
    const showPrivate = price_in_cents > 0;

    return (
      <li
        className={
          classNames(
            "repo",
            {"repo--active": active}
          )
        }
      >
        <div className="repo-name">
          {name}
        </div>

        { showPrivate &&
          <span className="badge margin-left-small">
            Private
          </span>
        }

        <div className="repo-activation-toggle">
          {this.renderButton()}
        </div>
      </li>
    );
  }
}

