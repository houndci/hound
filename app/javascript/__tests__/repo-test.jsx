import Repo from '../components/ReposContainer/components/Repo';

it('renders a repo appropriately', () => {
  const repo = {
    id: 1,
    name: "Test repo",
    owner: {
      id: 1
    }
  };

  const onRepoClicked = jest.fn();

  const wrapper = shallow(
    <Repo
      repo={repo}
      key={repo.id}
      onRepoClicked={onRepoClicked}
      isProcessingId={null}
    />
  );
  expect(wrapper).toMatchSnapshot();
});
