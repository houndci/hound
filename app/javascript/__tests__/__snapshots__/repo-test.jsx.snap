exports[`test renders a repo appropriately 1`] = `
<li
  className="repo">
  <div
    className="repo-name">
    Test repo
  </div>
  <div
    className="repo-activation-toggle">
    <RepoActivationButton
      isProcessingId={null}
      onRepoClicked={[Function]}
      repo={
        Object {
          "id": 1,
          "name": "Test repo",
          "owner": Object {
            "id": 1,
          },
        }
      } />
  </div>
</li>
`;
